sagemaker/pipeline/create_sm_pipeline.py

```diff
@@ -0,0 +1,693 @@
+        dist_part_processor = ScriptProcessor(
+            image_uri=args.aws_config.graphstorm_pytorch_cpu_image_url,
+            role=args.aws_config.execution_role,
+            instance_count=self.instance_count_param,
+            instance_type=self.graphconstruct_instance_type_param,
+            command=["python3"],
+            sagemaker_session=self.pipeline_session,
+            volume_size_in_gb=self.volume_size_gb_param,
+        )
+
+        partition_output = Join(
+            on="/",
+            values=[
+                self.output_subpath,
+                "gspartition",
+            ],
+        )
+
+        # Expected to be a list of strings
+        dist_part_arguments = [
+            "--graph-data-s3",
+            self.next_step_data_input,
+            "--metadata-filename",
+            args.partition_config.input_json_filename,
+            "--num-parts",
+            self.instance_count_param.to_string(),
+            "--output-data-s3",
+            partition_output,
+            "--partition-algorithm",
+            self.partition_algorithm_param,
+        ]
+
+        distpart_processor_args = dist_part_processor.run(
+            inputs=[],
+            outputs=[],
+            code=args.script_paths.dist_part_script,
+            arguments=dist_part_arguments,
+        )
+
+        dist_part_step = ProcessingStep(
+            name="DistPartition",
+            step_args=distpart_processor_args,  # type: ignore
+            cache_config=self.cache_config,
+        )
+
+        # Append dist_graph to input path, because it's added by GSPartition
+        self.next_step_data_input = Join(
+            on="/",
+            values=[
+                partition_output,
+                "dist_graph",
+            ],
+        )
+
+        return dist_part_step
+
+    def _create_gb_convert_step(self, args: PipelineArgs) -> ProcessingStep:
+        # Implementation for GraphBolt partition step
+        gb_part_processor = ScriptProcessor(
+            image_uri=args.aws_config.graphstorm_pytorch_cpu_image_url,
+            role=args.aws_config.execution_role,
+            instance_count=1,
+            instance_type=self.graphconstruct_instance_type_param,
+            command=["python3"],
+            sagemaker_session=self.pipeline_session,
+            volume_size_in_gb=self.volume_size_gb_param,
+        )
+
+        gb_convert_arguments = [
+            "--metadata-filename",
+            args.partition_config.output_json_filename,
+        ]
+
+        gb_convert_step = ProcessingStep(
+            name="GraphBoltConversion",
+            code=args.script_paths.gb_convert_script,
+            processor=gb_part_processor,
+            inputs=[
+                ProcessingInput(
+                    input_name="dist_graph_s3_input",
+                    destination="/opt/ml/processing/dist_graph/",
+                    source=self.next_step_data_input,
+                    s3_input_mode="File",
```

Comment:
Are we loading S3 files by our own script?

Reply:
Not for GBConvert, we use the SageMaker service to download the graph input at job launch time.

Reply:
Will it cause scalability issues?